app/src/main/java/science/apolline/view/activity/SplashScreen.kt

```diff
@@ -333,56 +328,60 @@ class SplashScreen : RootActivity(), AnkoLogger {
                 }
             }
 
-            mDetectedDevices[nameOrcode] = device
+            if (mDetectedDevices.containsKey(nameOrcode))
+                return
 
-            when (device.bluetoothClass.majorDeviceClass) {
+            else {
+                mDetectedDevices[nameOrcode] = device
 
-                BluetoothClass.Device.Major.AUDIO_VIDEO -> {
-                    onAddClick(circle_layout, nameOrcode, R.drawable.ic_device_bluetooth_audio_video, isCompatible, isBounded)
-                }
-                BluetoothClass.Device.Major.COMPUTER -> {
-                    onAddClick(circle_layout, nameOrcode, R.drawable.ic_device_bluetooth_computer, isCompatible, isBounded)
-                }
-                BluetoothClass.Device.Major.HEALTH -> {
-                    onAddClick(circle_layout, nameOrcode, R.drawable.ic_device_bluetooth_health, isCompatible, isBounded)
-                }
+                when (device.bluetoothClass.majorDeviceClass) {
 
-                BluetoothClass.Device.Major.IMAGING -> {
-                    onAddClick(circle_layout, nameOrcode, R.drawable.ic_device_bluetooth_imaging, isCompatible, isBounded)
-                }
+                    BluetoothClass.Device.Major.AUDIO_VIDEO -> {
+                        onAddClick(circle_layout, nameOrcode, R.drawable.ic_device_bluetooth_audio_video, isCompatible, isBounded)
```

Comment:
Looks like this code can be refactored to avoid multiple calls to the same method with a distinct constant

Reply:
can't touch this at the moment, it was there also in previous releases, this one will work as well. I can fix it for a futur release if that possible.